src/DocumentDB.ChangeFeedProcessor/FeedProcessing/ChangeFeedDocumentClientHealthDecorator.cs

```diff
@@ -0,0 +1,158 @@
+﻿// ----------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.  Licensed under the MIT license.
+// ----------------------------------------------------------------
+namespace Microsoft.Azure.Documents.ChangeFeedProcessor.FeedProcessing
+{
+    using System;
+    using System.Linq;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Documents.ChangeFeedProcessor.DataAccess;
+    using Microsoft.Azure.Documents.ChangeFeedProcessor.Monitoring;
+    using Microsoft.Azure.Documents.Client;
+    using Microsoft.Azure.Documents.Linq;
+
+    internal class ChangeFeedDocumentClientHealthDecorator : IChangeFeedDocumentClient
```

Comment:
Can we derive from DocumentClient and just override the method(s) we need?

Reply:
It cannot be the DocumentClient, its instance can be created by clients to the library and we do not want to replace it. IChangeFeedDocumentClient is used instead, we could derive from its implementation ChangeFeedDocumentClient but it would mean clients that have their own wrapper (used for telemetry as we do) would need to re-implement it themselves.

We could, however, call ExecuteNextAsync().AbortOnTimeout() from the processor itself and avoid wrappers at all. It would again mean that custom implementation of IPartitionProcessor will need to re-implement it.

Reply:
Yes, I meant deriving from ChangeFeedDocumentClient.

Reply:
Ok, sounds good to me, will do.